src/app/server/JointFabricDatastore.cpp

```diff
@@ -122,16 +122,166 @@ CHIP_ERROR JointFabricDatastore::RemoveNode(NodeId nodeId)
     return CHIP_ERROR_NOT_FOUND;
 }
 
-CHIP_ERROR JointFabricDatastore::RefreshNode(NodeId nodeId)
+CHIP_ERROR JointFabricDatastore::RefreshNode(NodeId nodeId, ReadOnlyBuffer<DataModel::EndpointEntry> endpointsList)
 {
     // 1. && 2.
     ReturnErrorOnFailure(SetNode(nodeId, Clusters::JointFabricDatastore::DatastoreStateEnum::kPending));
 
-    // 3. TODO: Read the PartsList of the Descriptor cluster from the Node.
+    // 3.
+
+    // cycle through endpointsList and add them to the endpoint entries
+    for (const auto & endpoint : endpointsList)
+    {
+        auto it =
+            std::find_if(mEndpointEntries.begin(), mEndpointEntries.end(),
+                         [&endpoint](const Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type & entry) {
+                             return entry.endpointID == endpoint.id;
+                         });
+        if (it == mEndpointEntries.end())
+        {
+            Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type newEntry;
+            newEntry.endpointID = endpoint.id;
+            newEntry.nodeID     = nodeId;
+            mEndpointEntries.push_back(newEntry);
+        }
+    }
+
+    // Remove EndpointEntries that are not in the endpointsList
+    for (auto it = mEndpointEntries.begin(); it != mEndpointEntries.end();)
+    {
+        if (it->nodeID == nodeId)
+        {
+            bool found = false;
+
+            for (const auto & endpoint : endpointsList)
+            {
+                if (it->endpointID == endpoint.id)
+                {
+                    found = true;
+                    break;
+                }
+            }
+
+            if (!found)
+            {
+                // If the endpoint is not in the endpointsList, we remove it
+                it = mEndpointEntries.erase(it);
+                continue;
+            }
+        }
+
+        ++it;
+    }
+
```

Comment:
```suggestion
     // TODO: read the Group List from the actual device
```

Reply:
Isn't the datastore already supposed to reflect this information? Or should this sync (if needed) be handled outside of the RefreshNode command?